File: GdPicture/WinForm/C#/Document Viewer/frmPrint.cs
using GdPicture14;
using System;
using System.Windows.Forms;

namespace Document_Viewer
{
    public partial class frmPrint
    {

        public enum PrintOrientation
        {
            AutoDetection,
            Portrait,
            Paysage
        }

        public enum PagesToPrint
        {
            All,
            Selection,
            Current,
            Range
        }

        public struct PrintSettings
        {
            public string Printer;
            public short Copies;
            public bool Collate;
            public PrintSizeOption PrintSize;
            public PrintAlignment PrintAlignment;
            public PrintOrientation Orientation;
            public PagesToPrint PagesToPrint;
            public string PageRange;
        }

        private PrintSettings _printSettings;
        private readonly GdViewer _owner;

        public PrintSettings printConfiguration
        {
            get
            {
                return _printSettings;
            }
        }

        private bool m_isSelectionPartial;
        private bool m_isEmptySelection;

        public frmPrint(GdViewer owner)
        {
            InitializeComponent();
            _owner = owner;
        }

        private void frmPrint_Load(object sender, EventArgs e)
        {
            cmbPrinterList.Items.Clear();
            for (int i = 1; i <= _owner.PrintGetPrintersCount(); i++)
            {
                string printerName = _owner.PrintGetPrinterName(i);
                cmbPrinterList.Items.Add(printerName);
                if (printerName == _owner.PrintGetActivePrinter())
                {
                    cmbPrinterList.SelectedIndex = cmbPrinterList.Items.IndexOf(printerName);
                }
            }

            cmbPrintSize.Items.Add("Actual size"); //0 = PrintSizeOptionActual
            cmbPrintSize.Items.Add("Fit to page"); //1 = PrintSizeOptionFit, default
            cmbPrintSize.Items.Add("Shrink oversized pages"); //2 = PrintSizeOptionShrinkOversized
            cmbPrintSize.SelectedIndex = 2;

            cmbPrintAlignment.Items.Add("Middle left"); //0 = PrintAlignmentMiddleLeft
            cmbPrintAlignment.Items.Add("Middle right"); //1 = PrintAlignmentMiddleRight
            cmbPrintAlignment.Items.Add("Middle center"); //2 = PrintAlignmentMiddleCenter
            cmbPrintAlignment.Items.Add("Default"); //3 = PrintAlignmentDefault
            cmbPrintAlignment.Items.Add("Top left"); //4 = PrintAlignmentTopLeft
            cmbPrintAlignment.Items.Add("Top right"); //5 = PrintAlignmentTopRight
            cmbPrintAlignment.Items.Add("Top center"); //6 = PrintAlignmentTopCenter
            cmbPrintAlignment.Items.Add("Bottom left"); //7 = PrintAlignmentBottomLeft
            cmbPrintAlignment.Items.Add("Bottom right"); //8 = PrintAlignmentBottomRight
            cmbPrintAlignment.Items.Add("Bottom center"); //9 = PrintAlignmentBottomCenter
            cmbPrintAlignment.SelectedIndex = 2;

            if (m_isEmptySelection)
            {
                rbPageRangeSelection.Enabled = false;
            }
            else
            {
                if (m_isSelectionPartial)
                {
                    rbPageRangeSelection.Checked = true;
                }
            }
            txtPageRangeStart.Enabled = false;
            txtPageRangeEnd.Enabled = false;
            txtPageRangeStart.Text = "1";
            txtPageRangeEnd.Text = _owner.PageCount.ToString();

            btnPrint.DialogResult = DialogResult.OK;
            btnCancel.DialogResult = DialogResult.Cancel;
        }

        private void btnPrinterProperties_Click(object sender, EventArgs e)
        {
            GdPictureStatus status = _owner.PrintShowPrinterSettingsDialog(this.Handle);
            if (status != GdPictureStatus.OK)
            {
                //todo: handle error
            }
        }

        private void cmbPrinterList_SelectedIndexChanged(object sender, EventArgs e)
        {
            _owner.PrintSetActivePrinter(cmbPrinterList.Text);
        }

        private void btnCancel_Click(object sender, EventArgs e)
        {
            this.Close();
        }

        private void rbPageRangeAll_CheckedChanged(object sender, EventArgs e)
        {
            rbPageRangeGeneric_CheckedChanged();
        }

        private void rbPageRangeSelection_CheckedChanged(object sender, EventArgs e)
        {
            rbPageRangeGeneric_CheckedChanged();
        }

        private void rbPageRangeCurrent_CheckedChanged(object sender, EventArgs e)
        {
            rbPageRangeGeneric_CheckedChanged();
        }

        private void rbPageRangePages_CheckedChanged(object sender, EventArgs e)
        {
            rbPageRangeGeneric_CheckedChanged();
        }

        private void rbPageRangeGeneric_CheckedChanged()
        {
            if (rbPageRangePages.Checked)
            {
                txtPageRangeStart.Enabled = true;
                txtPageRangeEnd.Enabled = true;
            }
            else
            {
                txtPageRangeStart.Enabled = false;
                txtPageRangeEnd.Enabled = false;
            }
        }

        private void btnPrint_Click(object sender, EventArgs e)
        {
            _printSettings.Printer = cmbPrinterList.Text;
            _printSettings.Copies = (short)numCopies.Value;
            _printSettings.Collate = chkCollate.Checked;
            _printSettings.PrintSize = (PrintSizeOption)cmbPrintSize.SelectedIndex;
            _printSettings.PrintAlignment = (PrintAlignment)cmbPrintAlignment.SelectedIndex;

            if (rbOrientationAutomatic.Checked)
            {
                _printSettings.Orientation = PrintOrientation.AutoDetection;
            }
            else if (rbOrientationPortrait.Checked)
            {
                _printSettings.Orientation = PrintOrientation.Portrait;
            }
            else if (rbOrientationLandscape.Checked)
            {
                _printSettings.Orientation = PrintOrientation.Paysage;
            }

            if (rbPageRangeAll.Checked)
            {
                _printSettings.PagesToPrint = PagesToPrint.All;
            }
            else if (rbPageRangeSelection.Checked)
            {
                _printSettings.PagesToPrint = PagesToPrint.Selection;
            }
            else if (rbPageRangeCurrent.Checked)
            {
                _printSettings.PagesToPrint = PagesToPrint.Current;
            }
            else if (rbPageRangePages.Checked)
            {
                _printSettings.PagesToPrint = PagesToPrint.Range;
                _printSettings.PageRange = txtPageRangeStart.Text + "-" + txtPageRangeEnd.Text;
            }
        }

    }
}

File: GdPicture/WinForm/vb.net/Annotations/frmPrint.vb
Option Explicit On
Option Infer On
Option Strict On

Imports System
Imports GdPicture14

Public Class frmPrint

    Enum PrintOrientation
        AutoDetection
        Portrait
        Paysage
    End Enum

    Enum PagesToPrint
        All
        Selection
        Current
        Range
    End Enum

    Public Structure PrintSettings
        Public Printer As String
        Public Copies As Short
        Public Collate As Boolean
        Public PrintSize As PrintSizeOption
        Public PrintAlignment As PrintAlignment
        Public Orientation As PrintOrientation
        Public PagesToPrint As PagesToPrint
        Public PageRange As String
    End Structure

    Private m_printSettings As PrintSettings
    Private ReadOnly m_owner As GdViewer

    Public ReadOnly Property printConfiguration() As PrintSettings
        Get
            Return m_printSettings
        End Get
    End Property


    Private ReadOnly m_isSelectionPartial As Boolean
    Private ReadOnly m_isEmptySelection As Boolean

    Public Sub New(ByVal owner As GdViewer)
        InitializeComponent()
        m_owner = owner
    End Sub

    Private Sub frmPrint_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
        cmbPrinterList.Items.Clear()
        For i As Integer = 1 To m_owner.PrintGetPrintersCount()
            Dim printerName As String = m_owner.PrintGetPrinterName(i)
            cmbPrinterList.Items.Add(printerName)
            If printerName = m_owner.PrintGetActivePrinter() Then
                cmbPrinterList.SelectedIndex = cmbPrinterList.Items.IndexOf(printerName)
            End If
        Next i

        cmbPrintSize.Items.Add("Actual size") '0 = PrintSizeOptionActual
        cmbPrintSize.Items.Add("Fit to page") '1 = PrintSizeOptionFit, default
        cmbPrintSize.Items.Add("Shrink oversized pages") '2 = PrintSizeOptionShrinkOversized
        cmbPrintSize.SelectedIndex = 2

        cmbPrintAlignment.Items.Add("Middle left") '0 = PrintAlignmentMiddleLeft
        cmbPrintAlignment.Items.Add("Middle right") '1 = PrintAlignmentMiddleRight
        cmbPrintAlignment.Items.Add("Middle center") '2 = PrintAlignmentMiddleCenter
        cmbPrintAlignment.Items.Add("Default") '3 = PrintAlignmentDefault
        cmbPrintAlignment.Items.Add("Top left") '4 = PrintAlignmentTopLeft
        cmbPrintAlignment.Items.Add("Top right") '5 = PrintAlignmentTopRight
        cmbPrintAlignment.Items.Add("Top center") '6 = PrintAlignmentTopCenter
        cmbPrintAlignment.Items.Add("Bottom left") '7 = PrintAlignmentBottomLeft
        cmbPrintAlignment.Items.Add("Bottom right") '8 = PrintAlignmentBottomRight
        cmbPrintAlignment.Items.Add("Bottom center") '9 = PrintAlignmentBottomCenter
        cmbPrintAlignment.SelectedIndex = 2

        If m_isEmptySelection Then
            rbPageRangeSelection.Enabled = False
        Else
            If m_isSelectionPartial Then
                rbPageRangeSelection.Checked = True
            End If
        End If
        txtPageRangeStart.Enabled = False
        txtPageRangeEnd.Enabled = False
        txtPageRangeStart.Text = "1"
        txtPageRangeEnd.Text = m_owner.PageCount.ToString()

        btnPrint.DialogResult = Windows.Forms.DialogResult.OK
        btnCancel.DialogResult = Windows.Forms.DialogResult.Cancel
    End Sub

    Private Sub btnPrinterProperties_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnPrinterProperties.Click
        Dim status As GdPictureStatus = m_owner.PrintShowPrinterSettingsDialog(Me.Handle)
        If status <> GdPictureStatus.OK Then
            'todo: handle error
        End If
    End Sub

    Private Sub cmbPrinterList_SelectedIndexChanged(ByVal sender As Object, ByVal e As EventArgs) Handles cmbPrinterList.SelectedIndexChanged
        m_owner.PrintSetActivePrinter(cmbPrinterList.Text)
    End Sub

    Private Sub btnCancel_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnCancel.Click
        Me.Close()
    End Sub

    Private Sub rbPageRangeAll_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles rbPageRangeAll.CheckedChanged
        rbPageRangeGeneric_CheckedChanged()
    End Sub

    Private Sub rbPageRangeSelection_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles rbPageRangeSelection.CheckedChanged
        rbPageRangeGeneric_CheckedChanged()
    End Sub

    Private Sub rbPageRangeCurrent_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles rbPageRangeCurrent.CheckedChanged
        rbPageRangeGeneric_CheckedChanged()
    End Sub

    Private Sub rbPageRangePages_CheckedChanged(ByVal sender As Object, ByVal e As EventArgs) Handles rbPageRangePages.CheckedChanged
        rbPageRangeGeneric_CheckedChanged()
    End Sub

    Private Sub rbPageRangeGeneric_CheckedChanged()
        If rbPageRangePages.Checked Then
            txtPageRangeStart.Enabled = True
            txtPageRangeEnd.Enabled = True
        Else
            txtPageRangeStart.Enabled = False
            txtPageRangeEnd.Enabled = False
        End If
    End Sub

    Private Sub btnPrint_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnPrint.Click
        m_printSettings.Printer = cmbPrinterList.Text
        m_printSettings.Copies = CShort(numCopies.Value)
        m_printSettings.Collate = chkCollate.Checked
        m_printSettings.PrintSize = CType(cmbPrintSize.SelectedIndex, PrintSizeOption)
        m_printSettings.PrintAlignment = CType(cmbPrintAlignment.SelectedIndex, PrintAlignment)

        If rbOrientationAutomatic.Checked Then
            m_printSettings.Orientation = PrintOrientation.AutoDetection
        ElseIf rbOrientationPortrait.Checked Then
            m_printSettings.Orientation = PrintOrientation.Portrait
        ElseIf rbOrientationLandscape.Checked Then
            m_printSettings.Orientation = PrintOrientation.Paysage
        End If

        If rbPageRangeAll.Checked Then
            m_printSettings.PagesToPrint = PagesToPrint.All
        ElseIf rbPageRangeSelection.Checked Then
            m_printSettings.PagesToPrint = PagesToPrint.Selection
        ElseIf rbPageRangeCurrent.Checked Then
            m_printSettings.PagesToPrint = PagesToPrint.Current
        ElseIf rbPageRangePages.Checked Then
            m_printSettings.PagesToPrint = PagesToPrint.Range
            m_printSettings.PageRange = txtPageRangeStart.Text & "-" & txtPageRangeEnd.Text
        End If
    End Sub

End Class